cpp/include/raft/linalg/svd.cuh

```diff
@@ -332,38 +340,52 @@ void svd_eig(
   raft::device_matrix_view<ValueType, IndexType, raft::col_major> V,
   std::optional<raft::device_matrix_view<ValueType, IndexType, raft::col_major>> U = std::nullopt)
 {
+  ValueType* left_sing_vecs_ptr = nullptr;
   if (U) {
     RAFT_EXPECTS(in.extent(0) == U.value().extent(0) && in.extent(1) == U.value().extent(1),
                  "U should have dimensions m * n");
+    left_sing_vecs_ptr = U.value().data_handle();
   }
-  RAFT_EXPECTS(in.extent(0) == V.extent(0) && in.extent(1) == V.extent(1),
+  RAFT_EXPECTS(in.extent(1) == V.extent(0) && in.extent(1) == V.extent(1),
                "V should have dimensions n * n");
   svdEig(handle,
          const_cast<ValueType*>(in.data_handle()),
          in.extent(0),
          in.extent(1),
          S.data_handle(),
-         U.value().data_handle(),
-         V.value().data_handle(),
+         left_sing_vecs_ptr,
+         V.data_handle(),
          U.has_value(),
          handle.get_stream());
 }
 
+template <typename ValueType, typename IndexType, typename UType>
+void svd_eig(const raft::handle_t& handle,
+             raft::device_matrix_view<const ValueType, IndexType, raft::col_major> in,
+             raft::device_vector_view<ValueType, IndexType> S,
+             raft::device_matrix_view<ValueType, IndexType, raft::col_major> V,
+             UType&& U = std::nullopt)
```

Comment:
Can you add a quick test-case to the googletests to make sure the optional `U` input is working? (This will also help us ensure we don't hit regressions should anything change in the future).